### PR TITLE
Fix Or&Nir bugs - first round

### PR DESCRIPTION
**Pull request by idoed**

### CODE REVIEW

File: src/com/company/Main.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
+package com.company;
+
+import com.company.model.Call;
+
+public class Main {
+
+    public static void main(String[] args) {
+        //I created 1 support panel as singleton because there'r not such a case 1 company has 2 team of it.
```
Comment (nirdev) [inline src/com/company/Main.java]: Stupid comments reloaded: 
this comment should be inside the singleton class not here, also I would write it like this:
"This class is a singleton due to the business logic of the program, we currently don't want any use case were their are to objects of this class"

File: src/com/company/SupportPanel.java [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+import java.util.ArrayList;
+import java.util.Deque;
+import java.util.LinkedList;
+import java.util.Queue;
+
+public class SupportPanel {
+    private static int i, mCallsId = 0;
+    private static ArrayList<Employee> mEmployeesLevel1, mEmployeesLevel2;
+    private static Employee mManager;
+    //Describe the number of people in each team- the num1 represent team level 1 and then second is level 2.
```
Comment (nirdev) [inline src/com/company/SupportPanel.java]: ????

File: src/com/company/SupportPanel.java [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+    private Deque<Call> mQueue = new LinkedList<>();
```
Comment (nirdev) [inline src/com/company/SupportPanel.java]: You should have a very good explanation why this is the collection you use, also write the explanation in a comment

File: src/com/company/SupportPanel.java [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+    private static SupportPanel instance = null;
+
+    public static SupportPanel getInstance() {
```
Comment (nirdev) [inline src/com/company/SupportPanel.java]: This is the place to note why you choose singleton pattern

File: src/com/company/SupportPanel.java [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+        if (instance == null) {
+            instance = new SupportPanel();
+        }
+        return instance;
+    }
+
+    //Constructor
```
Comment (nirdev) [inline src/com/company/SupportPanel.java]: explain why private

File: src/com/company/SupportPanel.java [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+    private SupportPanel() {
+    }
+
+    /**
+     * Creating new Employee's and put them in Arraylist According to their Level.
+     * The Method receiving 2 int params which represent the number of employees of each team.
+     */
+    public void settingUpEmployers(int mEmployeesLevel1Num, int mEmployeesLevel2Num) {
```
Comment (nirdev) [inline src/com/company/SupportPanel.java]: change the comment to be "green comment type" with param and return value and explain

File: src/com/company/SupportPanel.java [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+
+        mEmployeesLevel1 = new ArrayList<>();
+        mEmployeesLevel2 = new ArrayList<>();
+        //Setting the level 1 Employees
+        for (i = 0; i < mEmployeesLevel1Num; i++) {
+            Employee employee = new Employee(1, true);
+            mEmployeesLevel1.add(employee);
+        }//Setting the level 2 Employees
```
Comment (nirdev) [inline src/com/company/SupportPanel.java]: move the comment one line bellow the braces "{"

File: src/com/company/SupportPanel.java [hunk 8]
```diff
@@ -0,0 +1,126 @@
...
+        for (int j = 0; j < mEmployeesLevel2Num; j++) {
+            Employee employee = new Employee(2, true);
+            i++;
+            mEmployeesLevel2.add(employee);
+        }
+        //Setting the Manager
```
Comment (nirdev) [inline src/com/company/SupportPanel.java]: //Setting the Manager object*

File: src/com/company/SupportPanel.java [hunk 9]
```diff
@@ -0,0 +1,126 @@
...
+        mManager = new Employee(3, true);
+    }
+
+    /**
+     * Getting new Call and put it in our Deque.
+     *
+     * @param calls Receiving the Call Object from the Main activity.
+     */
+    public void HandleNewCall(Call calls) {
```
Comment (nirdev) [inline src/com/company/SupportPanel.java]: call not calls

File: src/com/company/SupportPanel.java [hunk 10]
```diff
@@ -0,0 +1,126 @@
...
+        calls.setSolve(false);
+        calls.setId(mCallsId);
+        mCallsId++;
+        mQueue.add(calls);
+        mQueue.add(calls);
+    }
+
+    public void AllocateCallsToEmployees() {
+        Call call = null;
+        if (mQueue != null) {
```
Comment (nirdev) [inline src/com/company/SupportPanel.java]: && mQueue length > 0

File: src/com/company/SupportPanel.java [hunk 11]
```diff
@@ -0,0 +1,126 @@
...
+        }
+        //After Polling from the Deque check if team1 tried to handle to call before.
+        if (call.getLevel() == 1) {
+            for (int j = 0; j < mEmployeesLevel1.size(); j++) {
+                Employee employee = mEmployeesLevel1.get(j);
+                if (employee.getAvailable()) {
+                    employee.setAvailable(false);
+                    //setting the Employee as busy
+                    mEmployeesLevel1.set(j, employee);
+                    /**If the User Solve The Problem i call the methods-
```
Comment (nirdev) [inline src/com/company/SupportPanel.java]: stupid comments revolution behind the scenes - you don't need all the //
also this looks messy

File: src/com/company/SupportPanel.java [hunk 12]
```diff
@@ -0,0 +1,126 @@
...
+
+    public void checkingNextGroup(Call call) {
+        //Setting the call for grade 2 so later on team 1 wont have to deal with it again.
+        call.setLevel(2);
+        for (int j = 0; j < mEmployeesLevel2.size(); j++) {
+            Employee employee = mEmployeesLevel2.get(j);
+            if (employee.getAvailable()) {
+                employee.setAvailable(false);
+                mEmployeesLevel2.set(j, employee);
+                /**If the user solve the Problem i call the methods-
```
Comment (nirdev) [inline src/com/company/SupportPanel.java]: stupid comments revolution behind the scenes II - you don't need all the //
also this looks messy

File: src/com/company/SupportPanel.java [hunk 13]
```diff
@@ -0,0 +1,126 @@
...
+            }
+        }
+        if (!call.getSolve()) {
+            //if the Call still unsolved than go to the next Group
+            checkingNextGroup(call);
+        }
+    }
+
+    public void checkingNextGroup(Call call) {
+        //Setting the call for grade 2 so later on team 1 wont have to deal with it again.
```
Comment (nirdev) [inline src/com/company/SupportPanel.java]: replace grade with level

Comment (nirdev) [reply src/com/company/SupportPanel.java]: won't*

File: src/com/company/model/Call.java [hunk 14]
```diff
@@ -0,0 +1,35 @@
+package com.company.model;
+
+public class Call {
+    //making default value of false because its represent the call problem.
+    private Boolean mSolve = false;
+    //making default value of 1 because the call go first to level 1 team
```
Comment (nirdev) [inline src/com/company/model/Call.java]: .

File: src/com/company/model/Employee.java [hunk 15]
```diff
@@ -0,0 +1,42 @@
+package com.company.model;
+
+/**
+ * Setting the employer Details, Level (1,2) , id , Available (True,False);
+ */
+public class Employee {
+
+    private int mLevel;
+    private static int mId = 0;
```
Comment (nirdev) [inline src/com/company/model/Employee.java]: write short comment on this unique Id mechanism

File: src/com/company/model/Employee.java [hunk 16]
```diff
@@ -0,0 +1,42 @@
...
+    private Boolean mAvailable;
+
+    public Employee(int mLevel, Boolean mAvailable) {
```
Comment (nirdev) [inline src/com/company/model/Employee.java]: (level,available)

File: src/com/company/model/Call.java [hunk 17]
```diff
@@ -0,0 +1,35 @@
...
+    int mId, mLevel = 1;
+
+    public Call() {
+    }
+
+    public int getLevel() {
+        return mLevel;
+    }
+
+    public void setLevel(int mLevel) {
```
Comment (nirdev) [inline src/com/company/model/Call.java]: int level

File: src/com/company/SupportPanel.java (hunk 6 above)
Comment (nirdev) [reply src/com/company/SupportPanel.java]: change the names to numberOfEmpolyeesLevel1

File: src/com/company/Main.java [hunk 18]
```diff
@@ -0,0 +1,16 @@
...
+        SupportPanel mySupportTeam = SupportPanel.getInstance();
```
Comment (nirdev) [inline src/com/company/Main.java]: move this to class level

File: src/com/company/Main.java [hunk 19]
```diff
@@ -0,0 +1,16 @@
...
+        mySupportTeam.settingUpEmployers(5, 2);
+        mySupportTeam.HandleNewCall(new Call());
+        mySupportTeam.AllocateCallsToEmployees();
```
Comment (nirdev) [inline src/com/company/Main.java]: call AllocateCallsToEmployees from inside HandleNewCall at the end

File: src/com/company/Main.java [hunk 20]
```diff
@@ -0,0 +1,16 @@
+package com.company;
+
+import com.company.model.Call;
+
+public class Main {
+
+    public static void main(String[] args) {
```
Comment (nirdev) [inline src/com/company/Main.java]: verify all method start with small letter and all method parameters don't have m.

File: src/com/company/SupportPanel.java [hunk 21]
```diff
@@ -0,0 +1,126 @@
...
+        calls.setId(mCallsId);
+        mCallsId++;
+        mQueue.add(calls);
+        mQueue.add(calls);
+    }
+
+    public void AllocateCallsToEmployees() {
+        Call call = null;
+        if (mQueue != null) {
+            call = mQueue.poll();
```
Comment (nirdev) [inline src/com/company/SupportPanel.java]: short comment

File: src/com/company/Main.java (hunk 1 above)
Comment (nirdev) [reply src/com/company/Main.java]: Also, add logs - each important method should have log with TAG, METHOD NAME and few words about what you're doing.
better to do this in a separate commit to prevent messy reviews

File: src/com/company/Main.java (hunk 20 above)
Comment (nirdev) [inline src/com/company/Main.java]: Also, add logs - each important method should have log with TAG, METHOD NAME and few words about what you're doing.
better to do this in a separate commit to prevent messy reviews

File: src/com/company/SupportPanel.java [hunk 22]
```diff
@@ -0,0 +1,126 @@
...
+        } else {
+            System.out.println("There are no calls");
+            return;
+        }
+        //After Polling from the Deque check if team1 tried to handle to call before.
```
Comment (nirdev) [inline src/com/company/SupportPanel.java]: handle the call

File: src/com/company/SupportPanel.java (hunk 4 above)
Comment (idoed) [reply src/com/company/SupportPanel.java]: done

File: src/com/company/SupportPanel.java (hunk 5 above)
Comment (idoed) [reply src/com/company/SupportPanel.java]: done

File: src/com/company/SupportPanel.java (hunk 13 above)
Comment (idoed) [reply src/com/company/SupportPanel.java]: done

File: src/com/company/model/Employee.java (hunk 15 above)
Comment (idoed) [reply src/com/company/model/Employee.java]: done

File: src/com/company/model/Employee.java (hunk 16 above)
Comment (idoed) [reply src/com/company/model/Employee.java]: done

File: src/com/company/SupportPanel.java (hunk 21 above)
Comment (idoed) [reply src/com/company/SupportPanel.java]: done

File: src/com/company/SupportPanel.java (hunk 22 above)
Comment (idoed) [reply src/com/company/SupportPanel.java]: done

File: src/com/company/SupportPanel.java (hunk 3 above)
Comment (idoed) [reply src/com/company/SupportPanel.java]: done

File: src/com/company/Main.java (hunk 1 above)
Comment (idoed) [reply src/com/company/Main.java]: done

File: src/com/company/CallsManager.java [hunk 23]
```diff
@@ -2,51 +2,54 @@
 
 import com.company.model.Call;
 import com.company.model.Employee;
+import sun.rmi.runtime.Log;
```
Comment (nirdev) [inline src/com/company/CallsManager.java]: bad import

Comment (nirdev) [reply src/com/company/CallsManager.java]: @idoed

File: src/com/company/SupportPanel.java (hunk 3 above)
Comment (nirdev) [reply src/com/company/SupportPanel.java]: I (not i , use capital)
"at the same time"

File: src/com/company/SupportPanel.java (hunk 7 above)
Comment (nirdev) [reply src/com/company/SupportPanel.java]: @idoed  fix that

File: src/com/company/SupportPanel.java (hunk 8 above)
Comment (nirdev) [reply src/com/company/SupportPanel.java]: @idoed  Add the word object at the end of the sentence

File: src/com/company/CallsManager.java [hunk 24]
```diff
@@ -2,51 +2,54 @@
...
         }//Setting the level 2 Employees
-        for (int j = 0; j < mEmployeesLevel2Num; j++) {
+        for (int j = 0; j < employeesLevel2Num; j++) {
             Employee employee = new Employee(2, true);
             i++;
             mEmployeesLevel2.add(employee);
         }
         //Setting the Manager
         mManager = new Employee(3, true);
+        System.out.println(TAG+" settingUpEmployers ,The Team is up and ready to handle the calls");
```
Comment (nirdev) [inline src/com/company/CallsManager.java]: TAG + "  - with spaces

File: src/com/company/SupportPanel.java (hunk 9 above)
Comment (nirdev) [reply src/com/company/SupportPanel.java]: @idoed  fix

File: src/com/company/CallsManager.java (hunk 24 above)
Comment (nirdev) [reply src/com/company/CallsManager.java]: @idoed
